R/get-param.R

```diff
@@ -0,0 +1,155 @@
+
+#' Return a formatted object for a given parameter
+#'
+#' @param .summary a `bbi_nonmem_summary` or `bbi_summary_list` object.
+#' @param .param the parameter(s) to retrieve and format. Any subset of `c("omega", "sigma", "theta")`
+#'
+#' @importFrom stats setNames
+#' @keywords internal
+get_param <- function(.summary, .param = c("omega", "sigma", "theta")){
+
+  if (inherits(.summary, NM_SUM_CLASS)) {
+    .summary <- list(.summary)
+    check_model_object_list(.summary, .mod_types = c(NM_SUM_CLASS))
```

Comment:
I'm confused by `check_model_object_list` is achieving here.  In the condition right above, you just did the same `inherits` check on `NM_SUM_CLASS` that `check_model_object_list` did underneath.  So isn't the failure unreachable?

https://github.com/metrumresearchgroup/bbr/blob/033899436217eb6a24083f97ae2063933b75a828/R/utils.R#L385-L386

Similar comment for the `check_model_object` call for the next condition.

Reply:
Yes I had this thought after merging the test_threads PR a while back (used in `check_run_times()`). I forget how I had it originally, but Seth and I came up with that together. I dont think either of us fully inspected what the `check_` functions were doing at that point (we probably assumed it did more than confirm the class). In the case of `check_run_times()` we wanted to ensure a model object, model list, model summary, or model summary list object was passed in, and in this case we only want to allow a model summary or model summary list. Because of classes we weren't able to use solely `check_model_object` or `check_model_object_list`.



Ill rework this to make it actually meaningful. Will likely ditch the check_ functions altogether

Reply:
Okay, to be clear, not a big deal, I was just scratching my head.

Reply:
No worries. Made a change that I think improves the logic. Will hold off on touching the other function.